diff --git a/includes/common.inc b/includes/common.inc
index 07373ac47e..0e444cbab9 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -4714,8 +4714,14 @@ function drupal_process_attached($elements, $group = JS_DEFAULT, $dependency_che
       // In some cases, the first parameter ($data) is an array. Arrays can't be
       // passed as keys in PHP, so we have to get $data from the value array.
       if (is_numeric($data)) {
-        $data = $options['data'];
-        unset($options['data']);
+        // In some cases, $options is NULL, which results in a Notice here.
+        if (isset($options['data'])) {
+          $data = $options['data'];
+          unset($options['data']);
+        }
+        else {
+          $data = NULL;
+        }
       }
       // Apply the default group if it isn't explicitly given.
       if (!isset($options['group'])) {
